reference/test/matrix/coo_kernels.cpp

```diff
@@ -98,6 +98,48 @@ class Coo : public ::testing::Test {
 TYPED_TEST_CASE(Coo, gko::test::ValueIndexTypes);
 
 
+TYPED_TEST(Coo, ConvertsToPrecision)
+{
+    using ValueType = typename TestFixture::value_type;
+    using IndexType = typename TestFixture::index_type;
+    using OtherType = typename gko::next_precision<ValueType>;
+    using Coo = typename TestFixture::Mtx;
+    using OtherCoo = gko::matrix::Coo<OtherType, IndexType>;
+    auto tmp = OtherCoo::create(this->exec);
+    auto res = Coo::create(this->exec);
+    // If OtherType is more precise: 0, otherwise r
+    auto residual = r<OtherType>::value < r<ValueType>::value
+                        ? gko::remove_complex<ValueType>{0}
+                        : gko::remove_complex<OtherType>{r<OtherType>::value};
```

Comment:
nit: The `r` struct already removes complex types (see definition [here](https://github.com/ginkgo-project/ginkgo/blob/develop/core/test/utils.hpp#L90))
```suggestion
                        : r<OtherType>::value;
```

Reply:
The point of these lines was actually to make sure that both branches of the ternary operator have the same type, so this was supposed to be
```suggestion
                        : gko::remove_complex<ValueType>{r<OtherType>::value};
```
Do you think that's okay? Or overly cautious?

Reply:
That is fine for me as well.